tests/test_mssqlcliclient.py

```diff
@@ -149,81 +164,55 @@ def drop_entities(mssqlcli_client):
-            for rows, _, _, query, is_error in client.execute_query(multi_statement_query):
-                if query == u"select 'Morning' as [Name] UNION ALL select 'Evening'":
-                    self.assertTrue(len(rows), 2)
-                else:
-                    self.assertTrue(len(rows), 1)
-
-            for rows, _, _, query, is_error in \
-                    client.execute_query(multi_statement_query2):
-                if query == u"select 1":
-                    self.assertTrue(len(rows) == 1)
-                else:
-                    self.assertTrue(is_error)
-
-            for rows, _, _, query, is_error in \
-                    client.execute_query(multi_statement_query3):
-                if query == u"select 2":
-                    self.assertTrue(len(rows) == 1)
-                else:
-                    self.assertTrue(is_error)
-
-        finally:
-            shutdown(client)
-
-    def test_stored_proc_multiple_result_sets(self):
+    def test_stored_proc_multiple_result_sets(client):
         """
             Verify the results of running a stored proc with multiple result sets
         """
-        try:
-            client = create_mssql_cli_client()
-            create_stored_proc = u"CREATE PROC sp_mssqlcli_multiple_results " \
-                          u"AS " \
-                          u"BEGIN " \
-                          u"SELECT 'Morning' as [Name] UNION ALL select 'Evening' " \
-                          u"SELECT 'Dawn' as [Name] UNION ALL select 'Dusk' " \
-                          u"UNION ALL select 'Midnight' " \
-                          u"END"
-            exec_stored_proc = u"EXEC sp_mssqlcli_multiple_results"
-            del_stored_proc = u"DROP PROCEDURE sp_mssqlcli_multiple_results"
+        create_stored_proc = u"CREATE PROC sp_mssqlcli_multiple_results " \
+                        u"AS " \
+                        u"BEGIN " \
+                        u"SELECT 'Morning' as [Name] UNION ALL select 'Evening' " \
+                        u"SELECT 'Dawn' as [Name] UNION ALL select 'Dusk' " \
+                        u"UNION ALL select 'Midnight' " \
+                        u"END"
+        exec_stored_proc = u"EXEC sp_mssqlcli_multiple_results"
+        del_stored_proc = u"DROP PROCEDURE sp_mssqlcli_multiple_results"
 
+        try:
             list(client.execute_query(create_stored_proc))
             row_counts = []
             for rows, _, _, _, _ in client.execute_query(exec_stored_proc):
                 row_counts.append(len(rows))
-            self.assertTrue(row_counts[0] == 2)
-            self.assertTrue(row_counts[1] == 3)
-            list(client.execute_query(del_stored_proc))
+            assert row_counts[0] == 2
+            assert row_counts[1] == 3
         finally:
-            shutdown(client)
+            list(client.execute_query(del_stored_proc))
+
+
+class TestMssqlCliClientMultipleStatement(MssqlCliClient):
```

Comment:
This method was refactored for improved testing logic and readability--`if` statements were removed in favor of parametrized data.